src/azure-cli/azure/cli/command_modules/storage/operations/account.py

```diff
@@ -70,8 +73,30 @@ def create_storage_account(cmd, resource_group_name, account_name, sku=None, loc
 
     if custom_domain:
         params.custom_domain = CustomDomain(name=custom_domain, use_sub_domain=None)
+
+    # Encryption
     if encryption_services:
         params.encryption = Encryption(services=encryption_services)
+
+    if encryption_key_source is not None:
+        params.encryption.key_source = encryption_key_source
+
+    if params.encryption.key_source and params.encryption.key_source == "Microsoft.Keyvault":
+        if params.encryption.key_vault_properties is None:
+            KeyVaultProperties = cmd.get_models('KeyVaultProperties')
+            params.encryption.key_vault_properties = KeyVaultProperties(key_name=encryption_key_name,
+                                                                        key_vault_uri=encryption_key_vault,
+                                                                        key_version=encryption_key_version)
```

Comment:
Shall we add
```Python
    else:
        if any([encryption_key_name, encryption_key_vault, encryption_key_version]):
            raise ValueError(
                'Specify `--encryption-key-source=Microsoft.Keyvault` to configure key vault properties.')
```
like what `update_storage_account` does

Reply:
Actually it is by design.
In create scenario, we already have validator for input value as you can see below
https://github.com/Azure/azure-cli/blob/289c23826ee3d7956b48cfb2b9eba0bb6fcc5689/src/azure-cli/azure/cli/command_modules/storage/_validators.py#L488-L497

But in update scenario, we need to consider existing parameter values, so we add an addition check here.